apple/Sources/FerrostarMapLibreUI/Views/DynamicallyOrientingNavigationView.swift

```diff
@@ -69,28 +64,44 @@ public struct DynamicallyOrientingNavigationView<
         self.bottomTrailing = bottomTrailing()
 
         _camera = camera
-        _snappedZoom = snappedZoom
-        _useSnappedCamera = useSnappedCamera
+        self.navigationCamera = navigationCamera
     }
 
     public var body: some View {
-        switch orientation {
-        case .landscapeLeft, .landscapeRight:
-            Text("TODO")
-        default:
-            PortraitNavigationView(
-                styleURL: styleURL,
-                navigationState: navigationState,
-                camera: $camera,
-                snappedZoom: $snappedZoom,
-                useSnappedCamera: $useSnappedCamera,
-                onTapExit: onTapExit,
-                makeMapContent: { userLayers },
-                topCenter: { topCenter },
-                topTrailing: { topTrailing },
-                midLeading: { midLeading },
-                bottomTrailing: { bottomTrailing }
-            )
+        Group {
+            switch orientation {
+            case .landscapeLeft, .landscapeRight:
+                LandscapeNavigationView(
+                    styleURL: styleURL,
+                    camera: $camera,
+                    navigationCamera: navigationCamera,
+                    navigationState: navigationState,
+                    onTapExit: onTapExit,
+                    makeMapContent: { userLayers },
+                    topCenter: { topCenter },
+                    topTrailing: { topTrailing },
+                    midLeading: { midLeading },
+                    bottomTrailing: { bottomTrailing }
+                )
+            default:
+                PortraitNavigationView(
+                    styleURL: styleURL,
+                    camera: $camera,
+                    navigationCamera: navigationCamera,
+                    navigationState: navigationState,
+                    onTapExit: onTapExit,
+                    makeMapContent: { userLayers },
+                    topCenter: { topCenter },
```

Comment:
Something about the way we are doing this transition isn't quite right. I think part of the issue is that we end up recreating the underlying map view. This isn't necessarily bad if we can bootstrap the new map view to the correct state, but right now there are a t least two major issuses:

1. Location updates in MapLibre lag (somewhat by design); this isn't normally an issue, but it prevents quickly re-tracking the user (this is the sort of thing that makes me think Meta is right with obsessing over low-level camera control haha).
2. The initial camera isn't preserved at all (we rely on MapLibre's location tracking, but that's not going to work given the first issue). The map is reset at null island and then tries to fly to the user. We needed to fix this anyways with the plain portrait view, but switching orientations really highlights the problem.

Reply:
This is much improved with a singular navigation map view. It required a decent bit of rework, but in testing looks pretty sharp now